cmd/elrond/ring.go

```diff
@@ -0,0 +1,337 @@
+			return errors.Wrap(err, "provided server address not a valid address")
+		}
+
+		client := model.NewClient(serverAddress)
+
+		ringID, _ := command.Flags().GetString("ring")
+		name, _ := command.Flags().GetString("name")
+		priority, _ := command.Flags().GetInt("priority")
+		installationGroup, _ := command.Flags().GetString("installation-group")
+		soakTime, _ := command.Flags().GetInt("soak-time")
+		image, _ := command.Flags().GetString("image")
+		version, _ := command.Flags().GetString("version")
+
+		request := &model.UpdateRingRequest{
+			Name:              name,
+			Priority:          priority,
+			InstallationGroup: installationGroup,
+			SoakTime:          soakTime,
+			Image:             image,
+			Version:           version,
+		}
+
+		dryRun, _ := command.Flags().GetBool("dry-run")
+		if dryRun {
+			err := printJSON(request)
+			if err != nil {
+				return errors.Wrap(err, "failed to print API request")
+			}
+
+			return nil
+		}
+
+		ring, err := client.UpdateRing(ringID, request)
+		if err != nil {
+			return errors.Wrap(err, "failed to update ring")
+		}
+
+		if err = printJSON(ring); err != nil {
+			return errors.Wrap(err, "failed to print ring response")
+		}
+
+		return nil
+	},
+}
+
+var ringReleaseCmd = &cobra.Command{
+	Use:   "release",
+	Short: "Release an elrond ring.",
+	RunE: func(command *cobra.Command, args []string) error {
+		command.SilenceUsage = true
+
+		serverAddress, _ := command.Flags().GetString("server")
+		if _, err := url.Parse(serverAddress); err != nil {
+			return errors.Wrap(err, "provided server address not a valid address")
+		}
+
+		client := model.NewClient(serverAddress)
+		ringID, _ := command.Flags().GetString("ring")
+		image, _ := command.Flags().GetString("image")
+		version, _ := command.Flags().GetString("version")
+
+		request := &model.ReleaseRingRequest{
+			Image:   image,
+			Version: version,
+		}
+
+		dryRun, _ := command.Flags().GetBool("dry-run")
+		if dryRun {
+			err := printJSON(request)
+			if err != nil {
+				return errors.Wrap(err, "failed to print API request")
+			}
+
+			return nil
+		}
+
+		ring, err := client.ReleaseRing(ringID, request)
+		if err != nil {
+			return errors.Wrap(err, "failed to release a ring")
+		}
+
+		if err = printJSON(ring); err != nil {
+			return errors.Wrap(err, "failed to print ring release response")
+		}
+
+		return nil
+	},
+}
+
+var ringDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete a ring.",
+	RunE: func(command *cobra.Command, args []string) error {
+		command.SilenceUsage = true
+
+		serverAddress, _ := command.Flags().GetString("server")
+		if _, err := url.Parse(serverAddress); err != nil {
+			return errors.Wrap(err, "provided server address not a valid address")
+		}
+
+		client := model.NewClient(serverAddress)
+
+		ringID, _ := command.Flags().GetString("ring")
+
+		err := client.DeleteRing(ringID)
+		if err != nil {
+			return errors.Wrap(err, "failed to delete ring")
```

Comment:
Could you return the `ringID` in all the relevant errors please?

Reply:
Done, thanks